chef_postgres/libraries/pg_config.rb

```diff
@@ -155,25 +154,15 @@ def maintenance_work_memory
       def wal_keep_segments
         # Convert to modifier % of available disk space on log drive in MB, then divide by 16MB
         # If log location becomes configurable, the drive needs to also become configurable
-
+        puts node["filesystem"]["by_device"]
         modifier = { web: 0.50,
                      oltp: 0.50,
                      dw: 0.50,
                      mixed: 0.50,
                      desktop: 0.25,
         }.fetch(workload)
-
-        (kb_available.to_f * modifier / 1024 / 16).round
-      end
-
-      def kb_available
-        return node["filesystem"][mount_drive]["kb_available"] unless node["filesystem"][mount_drive].nil?
-
-        kb = 0
-        node["filesystem"].each do |key, value|
-          kb = value['kb_available'].to_i if value['kb_available'].to_i > kb
-        end
-        kb
+          (node["filesystem"]["by_device"]["/dev/nvme0n1p1"]["kb_available"].to_f * modifier / 1024 / 16).round
+          #(node["filesystem"]["by_device"]["/dev/xvda1"]["kb_available"].to_f * modifier / 1024 / 16).round
```

Comment:
Shouldn't the "/dev/nvme0n1p1" be a variable?

Reply:
Lets remove the comment

Reply:
that was one thing i was hoping to discuss how we should deal with that. it wasn't changing before but now going to the new m5's it is.  so yes I'll have to migrate it to a variable that gets passed in.